packages/teams-js/src/internal/communication.ts

```diff
@@ -614,40 +683,47 @@ function handleParentMessage(evt: DOMMessageEvent): void {
 
   if ('id' in evt.data && typeof evt.data.id === 'number') {
     // Call any associated Communication.callbacks
-    const message = evt.data as MessageResponse;
-    const callback = CommunicationPrivate.callbacks[message.id];
-    logger('Received a response from parent for message %i', message.id);
-    if (callback) {
-      logger('Invoking the registered callback for message %i with arguments %o', message.id, message.args);
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      callback.apply(null, [...message.args, message.isPartialResponse]);
-
-      // Remove the callback to ensure that the callback is called only once and to free up memory if response is a complete response
-      if (!isPartialResponse(evt)) {
-        logger('Removing registered callback for message %i', message.id);
-        delete CommunicationPrivate.callbacks[message.id];
+    const serializedResponse = evt.data as SerializedMessageResponse;
+    const message: MessageResponse = deserializeMessageResponse(serializedResponse);
+    const callbackId = retrieveMessageUUIDFromResponse(message);
```

Comment:
>retrieveMessageUUIDFromResponse [](http://example.com/codeflow?start=23&length=31)

Should this log in the case where `retrieveMessageUUIDFromResponse` returns `undefined` so that when we diagnose issues we can be alerted to the fact that we're receiving messages we don't recognize?

Reply:
That's a good call, I'll add that now